api/types/role.go

```diff
@@ -48,6 +48,10 @@ const (
 
 // Match checks if the given role matches this filter.
 func (f *RoleFilter) Match(role *RoleV6) bool {
+	if f.SkipSystemRoles {
+		return !IsSystemResource(role)
```

Comment:
Hi there @gzdunek, I think this line might have introduced a bug (at least on v14).

Right now if you search for roles in the web UI (at least within the Edit User modal), it will never reach the search keywords part below, thus search will never return filtered results by keyword.

We found this regression on one of our production instances while trying to add a new role to a user, search is always returning the same list regardless of the keyword.

Reply:
That's an obvious bug 🤦 
I shouldn't have changed the checks order.

Sorry about that, I will send a PR with a fix soon.
Thanks for reaching out.